cumulusci/tasks/bulkdata.py

```diff
@@ -797,6 +800,324 @@ def _sqlite_dump(self):
                 f.write(line + "\n")
 
 
+class GenerateMapping(BaseSalesforceApiTask):
+    task_docs = """
+    Generate a mapping file for use with the `extract_dataset` and `load_dataset` tasks.
+    This task will examine the schema in the specified org and attempt to infer a
+    mapping suitable for extracting data in packaged and custom objects as well as
+    customized standard objects.
+
+    Mappings cannot include reference cycles - situations where Object A refers to B,
+    and B also refers to A. Mapping generation will fail for such data models; to
+    resolve the issue, specify the `ignore` option with the name of one of the
+    involved lookup fields to suppress it. `ignore` can be specified as a list in
+    `cumulusci.yml` or as a comma-separated string at the command line.
+
+    In most cases, the mapping generated will need minor tweaking by the user. Note
+    that the mapping omits features that are not currently well supported by the
+    `extract_dataset` and `load_dataset` tasks, including self-lookups and references to
+    the `User` object.
+    """
+
+    task_options = {
+        "path": {"description": "Location to write the mapping file", "required": True},
+        "namespace_prefix": {"description": "The namespace prefix to use"},
+        "ignore": {
+            "description": "Object API names, or fields in Object.Field format, to ignore"
+        },
+    }
+
+    core_fields = ["Id", "Name", "FirstName", "LastName"]
+
+    def _init_options(self, kwargs):
+        super(GenerateMapping, self)._init_options(kwargs)
+        if "namespace_prefix" not in self.options:
+            self.options["namespace_prefix"] = ""
+
+        if self.options["namespace_prefix"] and not self.options[
+            "namespace_prefix"
+        ].endswith("__"):
+            self.options["namespace_prefix"] += "__"
+
+        if "ignore" in self.options:
+            self.options["ignore"] = process_list_arg(self.options["ignore"])
+        else:
+            self.options["ignore"] = []
```

Comment:
This could be a one-liner:

    self.options["ignore"] = process_list_arg(self.options.get("ignore", []))